Sources/Sentry/Public/SentrySpan.h

```diff
@@ -1,9 +1,93 @@
-#import <Foundation/Foundation.h>
+#import "SentrySpanContext.h"
+#import <Sentry/Sentry.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+@class SentryTransaction, SentrySpanId, SentryId, SentryHub;
+
 NS_SWIFT_NAME(Span)
-@protocol SentrySpan <NSObject>
+@interface SentrySpan : SentrySpanContext
+SENTRY_NO_INIT
+
+/**
+ * The timestamp of which the span ended.
+ */
+@property (nullable, nonatomic, strong) NSDate *timestamp;
+
+/**
+ * The start time of the span.
+ */
+@property (nullable, nonatomic, strong) NSDate *startTimestamp;
+
+/**
+ * An arbitrary mapping of additional metadata of the span.
+ */
+@property (nullable, readonly) NSDictionary<NSString *, id> *extras;
+
+/**
+ * Whether the span is finished.
+ */
+@property (readonly) bool isFinished;
+
+/**
+ * Init a SentrySpan with given transaction, traceId, parentSpanId and hub.
+ *
+ * @param transaction The transaction associated with this span.
+ * @param traceId Determines which trace the Span belongs to.
+ * @param parentId Id of a parent span.
+ *
+ * @return SentrySpan
+ */
+- (instancetype)initWithTransaction:(SentryTransaction *)transaction
+                            traceId:(SentryId *)traceId
+                        andParentId:(SentrySpanId *)parentId;
+
+/*
+ Removing SpanContext initializers
+ */
+- (instancetype)initWithSampled:(BOOL)sampled NS_UNAVAILABLE;
```

Comment:
`m`: Why do we have the initializer in `SpanContext` when we need to remove it here? @marandaneto is this intended?

Reply:
Because objc exposes the initializer of the base class. Java and .NET uses Interfaces, so they don't have this problem, although they don't expose de base class constructor.

Reply:
I didn't find documentation condemning hiding an initializer, I agree it is kind ugly (but fast) solution. 
We can solve the need to hide initializer in two ways. Using protocols, thus losing the benefits of inheritance, or by creating a third class (maybe Trace) with all basic implementations, and both SpanContext and Span inherit it. I prefer the latter.

Reply:
Wait. This seems weird to me. Why does `Span` inherit from `SpanContext` @marandaneto? Shouldn't `Span` have `SpanContext` as a property, in other words, use composition over inheritance? I know it's like that in Java, but this doesn't make sense to me.

Reply:
this is prior to my time working on performance.
.NET does not inherit from SpanContext but duplicates all its properties.

Java inherits from SpanContext, so a Span class reuses it, if it could be composition, well yes.
a Transaction on Java indeed has context property (SpanContext), so does composition because it inherits from SentryBaseEvent, and you cant inherit from multiple classes, that's my hunch.

Reply:
I'm not entirely sure I've understood the problem here as the class is outdated (did commit/push)

Reply:
Then I propose a few changes for the whole API in Cocoa. I put them here https://github.com/getsentry/sentry-cocoa/discussions/936#discussioncomment-340994, so we don't block this PR with this discussion.